src/Generators/DataTablesMakeCommand.php

```diff
@@ -34,51 +38,91 @@ class DataTablesMakeCommand extends GeneratorCommand
+                ),
+                '--dom'     => $this->option('dom') ?: $this->laravel['config']->get(
+                    'datatables-buttons.generator.dom',
+                    'Bfrtip'
+                ),
+                '--table'   => $this->option('table'),
+            ]);
+        }
+    }
+
     /**
      * Build the class with the given name.
      *
-     * @param  string $name
+     * @param string $name
      * @return string
      */
     protected function buildClass($name)
     {
         $stub = parent::buildClass($name);
 
-        return $this->replaceModelImport($stub)
-                    ->replaceModel($stub)
-                    ->replaceColumns($stub)
-                    ->replaceAction($stub)
-                    ->replaceFilename($stub);
+        $this->replaceModelImport($stub)
+             ->replaceModel($stub)
+             ->replaceBuilder($stub)
+             ->replaceColumns($stub)
+             ->replaceButtons($stub)
+             ->replaceDOM($stub)
+             ->replaceTableId($stub)
+             ->replaceAction($stub)
+             ->replaceFilename($stub);
+
+        return $stub;
     }
 
     /**
-     * Replace model name.
+     * Replace the filename.
      *
      * @param string $stub
-     * @return \Yajra\DataTables\Generators\DataTablesMakeCommand
+     * @return $this
      */
-    protected function replaceModel(&$stub)
+    protected function replaceFilename(&$stub)
     {
-        $model = explode('\\', $this->getModel());
-        $model = array_pop($model);
-        $stub  = str_replace('ModelName', $model, $stub);
+        $stub = str_replace(
+            'DummyFilename', preg_replace('#datatable$#i', '', $this->getNameInput()), $stub
+        );
 
         return $this;
     }
 
     /**
-     * Get model name to use.
+     * Replace the action.
+     *
+     * @param string $stub
+     * @return \Yajra\DataTables\Generators\DataTablesMakeCommand
      */
-    protected function getModel()
+    protected function replaceAction(&$stub)
     {
-        $name           = $this->getNameInput();
-        $rootNamespace  = $this->laravel->getNamespace();
-        $model          = $this->option('model') || $this->option('model-namespace');
-        $modelNamespace = $this->option('model-namespace') ? $this->option('model-namespace') : $this->laravel['config']->get('datatables-buttons.namespace.model');
+        $stub = str_replace(
+            'DummyAction', $this->getAction(), $stub
+        );
 
-        return $model
-            ? $rootNamespace . '\\' . ($modelNamespace ? $modelNamespace . '\\' : '') .  Str::singular($name)
-            : $rootNamespace . '\\User';
+        return $this;
+    }
+
+    /**
+     * Set the action view to be used.
+     *
+     * @return string
+     */
+    protected function getAction()
+    {
+        return $this->option('action') ? $this->option('action') : Str::lower($this->getNameInput()) . '.action';
```

Comment:
This can be simplified to `return $this->option('action') ?: Str::lower($this->getNameInput()) . '.action';`.